### ISSUE TITLE
Make Cannon Director feature work with Cannons plugin too

### ISSUE THREAD

**Issue by eirikh1996** [APDevTeam/Movecraft#188]
**Is your feature request related to a problem? Please describe.**
Currently, cannon directors only work with TNT cannons, but not with cannons provided by the Cannons plugin

**Describe the solution you'd like**
I would like to have Cannon directors work with the Cannons plugin too

**Comment by cccm5** [comment]
I don't think this is something that should be added directly to Movecraft - in fact I don't think any of the strictly combat oriented features should be in the main plugin. I plan on spiting off the combat portion to an addon that uses the API once the Kotlin version is released, so until than I don't think this should be a priority.

**Comment by FrozenBeard** [comment]
There is some form of Cannons integration in the existing version of movecraft as it makes the plugin VERY powerful, does not matter if you have a space server or a pirate server cannons are entirely custom and should definitely be the main plugin that Movecraft supports.
From memory, I think one of the plugins took weight of cannons into consideration for craft movement.
DerPavlov is amazing and very welcoming, please dont dismiss such an amazing plugin! It used to have a hook into it.

**Comment by TylerS1066** [comment]
> please dont dismiss such an amazing plugin!

I think cc is trying to say that the Movecraft plugin itself has become very bloated with features that aren't "core" to the functionality of Movecraft itself, and as such it would be best to split them off into separate plugins.

We should probably discuss what will stay in Movecraft core, and what will be moved into separate plugins so that we can certify that the API will do everything needed.

**Comment by CharlesLitton** [comment]
So have we decided to split the combat portion of movecraft off and just return to the core function of movecraft, which is moving ships? or are some combat features going to stay?

**Comment by eirikh1996** [comment]
@CharlesLitton The plan is to have a core Movecraft plugin, which will be about moving crafts built with Minecraft blocks. The additional stuff in Movecraft, such as repair, directors and warfare, will be split off as add-on plugins

For reference, you can look at how the Essentials developers did it with a core plugin and extra additions. It will be smth like that
